primap2/pm2io/_data_reading.py

```diff
@@ -1289,26 +1340,97 @@ def harmonize_units(
+            try:
+                native_unit = "Gg " + basic_entity + " / yr"
+                # print(f"Testing conversion from {ureg[unit_fallback].units} to "
+                #       f"{ureg[native_unit].units} for {basic_entity}.")
+                if gwp_to_use:
+                    if ureg(unit_fallback).is_compatible_with(
+                        ureg[native_unit], gwp_to_use
+                    ):
+                        unit_to = native_unit
+                        entity_conv = True
+                    else:
+                        unit_to = unit_fallback
+                        entity_conv = False
+                else:
+                    if ureg(unit_fallback).is_compatible_with(ureg[native_unit]):
+                        unit_to = native_unit
+                        entity_conv = False
+                    else:
+                        unit_to = unit_fallback
+                        entity_conv = False
+            except pint.UndefinedUnitError:
+                # we have a gas basket or something unknown, so no conversion to native
+                # unit
+                # print(f"Exception occured for entity {entity}")
+                unit_to = unit_fallback
+                entity_conv = False
+                native_unit = "None"
+
+            # print(f"unit_to is {unit_to}, unit_fallback is {unit_fallback}, and
+            # native_unit is {native_unit}")
+            # print(f"Using {unit_to} as target unit and entity_conv = {entity_conv}")
+
+            # if len(units_this_entity) > 1:
+            for unit in units_this_entity:
+                if unit != unit_to:
+                    # print(f"Working on unit {unit}")
+                    unit_pint = ureg[unit]
+                    # could add a try except block here to throw and log an error or add
+                    # error info in DF instead of crashing
+                    if gwp_to_use:
+                        with ureg.context(gwp_to_use):
+                            unit_pint = unit_pint.to(unit_to)
+                    else:
+                        unit_pint = unit_pint.to(unit_to)
+                    # print(f"Pint unit is {unit_pint}")
+                    factor = unit_pint.magnitude
+                    # print(f"Converting with factor {factor} to unit {unit_to}")
+                    mask = (data[entity_col] == entity) & (data[unit_col] == unit)
+                    # print(data.loc[mask, data_cols])
+                    try:
+                        data.loc[mask, data_cols] *= factor
+                    except TypeError:
+                        # print(data.loc[mask, data_cols])
+                        temp = []
+                        for col in data_cols:
+                            temp = temp + list(data[col].unique())
+                        temp = list(set(temp))
+
+                        # print(temp)
+                        strs = [x for x in temp if isinstance(x, str)]
+                        print(strs)
```

Comment:
This is effectively a `pass` on `TypeError`, and the function will soldier on. Maybe at least re-raise?

Reply:
Yes, that looks unfinished. It should raise the error and give the information on the strings that caused it.

Reply:
I've added code to re-raise the error. This will not happen when using the official primap2 csv reading functions though, as they already remove the strings